src/pushi/base/apn.py

```
@@ -61,23 +61,101 @@ def send(self, app_id, event, json_d, invalid={}):
+            tokens.extend(_tokens)
+        tokens = set(tokens)
+        count = len(tokens)
+
+        # prints a logging message about the various (web) subscriptions
+        # that were found for the event that was triggered
+        self.logger.debug("Found %d APN subscription(s) for '%s'" % (count, root_event))
+
+        # delegates to the direct send method with resolved tokens
+        self.send_to_tokens(tokens, message, app=app, invalid=invalid)
+
+    def send_to_tokens(self, tokens, message, app=None, key_data=None, cer_data=None, sandbox=None, invalid={}):
```

Ruff (0.14.10)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)